c10/cuda/CUDAMacros.h

```diff
@@ -41,11 +41,11 @@
 /**
  * The maximum number of GPUs that we recognizes. Increasing this beyond the
  * initial limit of 16 broke Caffe2 testing, hence the ifdef guards.
- * This value cannot be more than 255 because our DeviceIndex is a uint8_t.
+ * This value cannot be more than 32767 because our DeviceIndex is a int16_t.
```

Comment:
Ho wow, I missed the fact that this was actually signed when I wrote this. Good update!